fm.c

```diff
@@ -453,90 +657,192 @@ void FM_HandleIO2(fm_t *chip)
-            chip->slot_sl[i][j][to] = SLOT_ROTATE(chip->slot_sl[i][j][from]);
-        // ssg eg
-        for (j = 0; j < 4; j++)
-            chip->slot_ssg_eg[i][j][to] = SLOT_ROTATE(chip->slot_ssg_eg[i][j][from]);
-    }
-    // channel registers
-
-    // fnum
-    for (j = 0; j < 11; j++)
-        chip->chan_fnum[j][to] = CH_ROTATE(chip->chan_fnum[j][from]);
-    // fnum ch3
-    for (j = 0; j < 11; j++)
-        chip->chan_fnum_ch3[j][to] = CH_ROTATE(chip->chan_fnum_ch3[j][from]);
-    // block
-    for (j = 0; j < 3; j++)
-        chip->chan_block[j][to] = CH_ROTATE(chip->chan_block[j][from]);
-    // block ch3
-    for (j = 0; j < 3; j++)
-        chip->chan_block_ch3[j][to] = CH_ROTATE(chip->chan_block_ch3[j][from]);
-    // connect
-    for (j = 0; j < 3; j++)
-        chip->chan_connect[j][to] = CH_ROTATE(chip->chan_connect[j][from]);
-    // fb
-    for (j = 0; j < 3; j++)
-        chip->chan_fb[j][to] = CH_ROTATE(chip->chan_fb[j][from]);
-    // pms
-    for (j = 0; j < 3; j++)
-        chip->chan_pms[j][to] = CH_ROTATE(chip->chan_pms[j][from]);
-    // ams
-    for (j = 0; j < 2; j++)
-        chip->chan_ams[j][to] = CH_ROTATE(chip->chan_ams[j][from]);
-    // pan
-    for (j = 0; j < 2; j++)
-        chip->chan_pan[j][to] = CH_ROTATE(chip->chan_pan[j][from]);
+    // multi registers
+    chip->slot_multi[0][0][sel] = SLOT_ROTATE(chip->slot_multi[0][0][from]);
+    chip->slot_multi[0][1][sel] = SLOT_ROTATE(chip->slot_multi[0][1][from]);
+    chip->slot_multi[0][2][sel] = SLOT_ROTATE(chip->slot_multi[0][2][from]);
+    chip->slot_multi[0][3][sel] = SLOT_ROTATE(chip->slot_multi[0][3][from]);
+    chip->slot_multi[1][0][sel] = SLOT_ROTATE(chip->slot_multi[1][0][from]);
+    chip->slot_multi[1][1][sel] = SLOT_ROTATE(chip->slot_multi[1][1][from]);
+    chip->slot_multi[1][2][sel] = SLOT_ROTATE(chip->slot_multi[1][2][from]);
+    chip->slot_multi[1][3][sel] = SLOT_ROTATE(chip->slot_multi[1][3][from]);
+
+    // dt registers
+    chip->slot_dt[0][0][sel] = SLOT_ROTATE(chip->slot_dt[0][0][from]);
+    chip->slot_dt[0][1][sel] = SLOT_ROTATE(chip->slot_dt[0][1][from]);
+    chip->slot_dt[0][2][sel] = SLOT_ROTATE(chip->slot_dt[0][2][from]);
+    chip->slot_dt[1][0][sel] = SLOT_ROTATE(chip->slot_dt[1][0][from]);
+    chip->slot_dt[1][1][sel] = SLOT_ROTATE(chip->slot_dt[1][1][from]);
+    chip->slot_dt[1][2][sel] = SLOT_ROTATE(chip->slot_dt[1][2][from]);
+
+    // tl registers
+    chip->slot_tl[0][0][sel] = SLOT_ROTATE(chip->slot_tl[0][0][from]);
+    chip->slot_tl[0][1][sel] = SLOT_ROTATE(chip->slot_tl[0][1][from]);
+    chip->slot_tl[0][2][sel] = SLOT_ROTATE(chip->slot_tl[0][2][from]);
+    chip->slot_tl[0][3][sel] = SLOT_ROTATE(chip->slot_tl[0][3][from]);
+    chip->slot_tl[0][4][sel] = SLOT_ROTATE(chip->slot_tl[0][4][from]);
+    chip->slot_tl[0][5][sel] = SLOT_ROTATE(chip->slot_tl[0][5][from]);
+    chip->slot_tl[0][6][sel] = SLOT_ROTATE(chip->slot_tl[0][6][from]);
+    chip->slot_tl[1][0][sel] = SLOT_ROTATE(chip->slot_tl[i][0][from]);

```

Comment:
`i`? should be 1, how did this even compile on your side lol

Reply:
old revision somehow made it from a `git stash`. ill have to correct this.